searchcore/src/apps/proton/proton.cpp

```diff
@@ -268,10 +281,9 @@ App::main(int argc, char **argv)
         setupSignals();
         setup_fadvise();
         FastOS_ThreadPool threadPool(128_Ki);
-        FNET_Transport transport(buildTransportConfig());
-        transport.Start(&threadPool);
-        startAndRun(threadPool, transport, argc, argv);
-        transport.ShutDown(true);
+        Transport transport(buildTransportConfig());
+        transport.transport().Start(&threadPool);
```

Comment:
consider doing Start in the wrapper constructor (if Start is not called or it fails, WaitFinished will block forever)